webapp/tests/forms/steps/lotse/test_confirmation.py

```diff
@@ -0,0 +1,139 @@
+from app.forms.steps.lotse.confirmation import StepSummary
+from app.forms.steps.lotse_multistep_flow_steps.confirmation_steps import StepConfirmation
+from app.model.form_data import MandatoryFieldMissingValidationError
+
+
+@pytest.mark.usefixtures('test_request_context')
+class TestStepSummary:
+    @pytest.fixture
+    def step(self):
+        step = StepSummary(endpoint='lotse')
+        return step
+
+    def test_if_complete_correct_not_set_then_fail_validation(self, step):
+        data = MultiDict({})
+        form = step.InputForm(formdata=data)
+        assert form.validate() is False
+
+    def test_if_complete_correct_set_then_succ_validation(self, step):
+        data = MultiDict({'confirm_complete_correct': True})
+        form = step.InputForm(formdata=data)
+        assert form.validate() is True
+
+    def test_set_section_steps_in_render_info(self, make_test_request_context, step):
+        expected_summary_session_steps = {}
+        with make_test_request_context(), \
+                patch("app.forms.flows.lotse_flow.LotseMultiStepFlow._get_overview_data",
+                      MagicMock(return_value=expected_summary_session_steps)), \
+                patch("app.forms.flows.lotse_flow.MandatoryFormData.parse_obj"):
+            step.handle()
+
+            assert step.render_info.additional_info['section_steps'] == expected_summary_session_steps
+
+    def test_if_data_missing_then_set_next_url_correct(self, make_test_request_context, step):
+        data_with_missing_fields = {'steuernummer': 'C3P0'}
+
+        with make_test_request_context(method='POST', stored_data=data_with_missing_fields), \
+                patch('app.forms.flows.lotse_flow.LotseMultiStepFlow._get_overview_data'):
+            step.handle()
+
+            assert step.render_info.next_url == LotseMultiStepFlow(endpoint='lotse').url_for_step(StepSummary.name)
+
+    def test_if_data_missing_then_call_get_overview_data_with_missing_fields(self, make_test_request_context):
+        data_with_missing_fields = {'steuernummer': 'C3P0'}
+        missing_fields = ['spacecraft', 'droids']
+
+        with make_test_request_context(method='POST', stored_data=data_with_missing_fields), \
+                patch('app.forms.flows.step_chooser.StepChooser.default_data', return_value=False), \
+                patch('app.forms.flows.lotse_flow.LotseMultiStepFlow._get_overview_data') as get_overview, \
+                patch('app.forms.flows.lotse_flow.LotseMultiStepFlow._validate_mandatory_fields',
+                      MagicMock(side_effect=MandatoryFieldMissingValidationError(missing_fields))):
+            step = LotseStepChooser().get_correct_step(StepSummary.name, update_data=True)
+            step.handle()
+
+            get_overview.assert_called_once_with(data_with_missing_fields, missing_fields)
+
+    def test_if_data_missing_then_flash_message_with_missing_fields_once(self, make_test_request_context):
```

Comment:
Maybe we can talk about this, because these should not be the normal interaction (a GET request should not set update_data to `True`). I can see the advantages of testing the interaction on that level. However, I am wondering if this is the correct level to do it then.